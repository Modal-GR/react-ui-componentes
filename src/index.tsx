/* Form */
export { Button } from "./components/Button";
export { Input } from "./components/Input";

/* Transition */
export * as Transition from "./components/Transitions";
import Loader from "./components/Loader";
export { Loader };

/* Boxs */
import Card from "./components/Card";
export { Card };

/* Content */
import ResponsiveImage from "./components/Image";
export { ResponsiveImage };
export { Flex, Grid } from "./components/Layout";
import Stars from "./components/Stars";
export { Stars };
