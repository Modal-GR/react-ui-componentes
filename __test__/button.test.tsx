import * as React from "react";
import Enzyme, { shallow, ShallowWrapper } from "enzyme";
import { create, ReactTestRenderer } from "react-test-renderer";

import { Button } from "../src/index";

describe("Run tests of component <Button />", () => {
  const mockCallBack = jest.fn();
  const wrapper = shallow(<Button onClickButton={mockCallBack}>Click</Button>);

  it("should render button", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toBeTruthy();
  });

  it("should render text correct", () => {
    expect(wrapper.contains("Click")).toBeTruthy();
  });

  // Write props test default and custom value

  // Write test condition

  it("should run event click", () => {
    wrapper.simulate("click");
    expect(mockCallBack).toHaveBeenCalled();
  });
});
