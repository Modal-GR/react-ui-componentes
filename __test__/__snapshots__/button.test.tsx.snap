// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Run tests of component <Button /> should render button 1`] = `
<styled.button
  backgroundColor="#073d92"
  borderRadius="30px"
  color="#fff"
  hoverType={
    Object {
      "ZoomIn": true,
    }
  }
  onClick={[MockFunction]}
>
  Click
</styled.button>
`;
